File: macos/Sources/GrokAuth.swift
import Foundation

/// App auth façade — OAuth tokens only ([OidcSession]).
final class GrokAuth {
    static let shared = GrokAuth()

    private init() {}

    var isSignedIn: Bool { OidcSession.shared.hasTokens }

    func clear() {
        OidcSession.shared.clear()
        // Remove leftover WebKit data from pre-OAuth builds
        Self.purgeLegacyWebKitData()
    }

    /// Call at launch so old cookie sessions never count as signed-in.
    func purgeLegacyArtifacts() {
        Self.purgeLegacyWebKitData()
    }

    private static func purgeLegacyWebKitData() {
        let fm = FileManager.default
        let home = fm.homeDirectoryForCurrentUser
        let candidates = [
            home.appendingPathComponent("Library/WebKit/com.blankspeaker.GRLD"),
            home.appendingPathComponent("Library/HTTPStorages/com.blankspeaker.GRLD"),
            home.appendingPathComponent("Library/HTTPStorages/com.blankspeaker.GRLD.binarycookies"),
            home.appendingPathComponent("Library/Caches/com.blankspeaker.GRLD")
        ]
        for url in candidates {
            try? fm.removeItem(at: url)
        }
    }
}

File: android/app/src/main/java/com/blankspeaker/grld/GrokAuth.kt
package com.blankspeaker.grld

import android.content.Context
import android.webkit.CookieManager

/**
 * App auth façade — OAuth tokens only ([OidcSession]).
 * Legacy SSO cookies / WebView sessions are not used.
 */
object GrokAuth {

    fun isSignedIn(context: Context): Boolean = OidcSession.hasTokens(context)

    fun clear(context: Context) {
        OidcSession.clear(context)
        // Wipe any leftover cookie jar from older builds
        wipeLegacyCookieStore()
        context.getSharedPreferences("grld_auth", Context.MODE_PRIVATE).edit().clear().apply()
    }

    /** One-shot cleanup of pre-OAuth cookie prefs (safe to call every launch). */
    fun purgeLegacyAuthArtifacts(context: Context) {
        context.getSharedPreferences("grld_auth", Context.MODE_PRIVATE).edit().clear().apply()
        wipeLegacyCookieStore()
    }

    private fun wipeLegacyCookieStore() {
        try {
            CookieManager.getInstance().removeAllCookies(null)
            CookieManager.getInstance().flush()
        } catch (_: Exception) { }
    }
}
